actions_helper/commands/deregister_task_definition.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional
+
+import click
+from botocore.client import BaseClient
+
+from actions_helper.outputs import CreateTaskDefinitionOutput
+from actions_helper.utils import set_error
+
+
+def deregister_task_definition(
+    ecs_client: BaseClient,
+    cluster: str,
+    service: str,
+    production_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    local_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
+    preflight_task_definition_output: Optional[CreateTaskDefinitionOutput] = None,
```

Comment:
```suggestion
    production_task_definition_output: Optional[CreateTaskDefinitionOutput],
    local_task_definition_output: Optional[CreateTaskDefinitionOutput],
    preflight_task_definition_output: Optional[CreateTaskDefinitionOutput],
```

No need for defaults here, from what I can tell.

Reply:
default is needed
Optional is just for typing, it is a type-hint to be used by IDEs and other static analysis tools, not a default argument

Reply:
We could confirm that default is not needed, will be removed.